docs/standard/unsafe-code/best-practices.md

```diff
@@ -0,0 +1,1006 @@
+
+### Recommendations
+
+1. ❌ DON'T write "branchless" conversions between integers and Booleans using unsafe code.
+2. ✔️ DO use ternary operators (or other branching logic) instead. Modern .NET JITs will optimize them effectively.
+3. ❌ DON'T read `bool` using unsafe APIs such as `Unsafe.ReadUnaligned` or `MemoryMarshal.Cast` if you don't trust the input. Consider using ternary operators or equality comparisons instead:
+
+```cs
+// Bad:
+bool b = Unsafe.ReadUnaligned<bool>(ref byteData);
+
+// Good:
+bool b = byteData[0] != 0;
+
+// Bad:
+ReadOnlySpan<byte> byteSpan = ReadDataFromNetwork();
+bool[] boolArray = MemoryMarshal.Cast<byte, bool>(byteSpan).ToArray();
+
+// Good:
+ReadOnlySpan<byte> byteSpan = ReadDataFromNetwork();
+bool[] boolArray = new bool[byteSpan];
+for (int i = 0; i < byteSpan.Length; i++) { boolArray[i] = byteSpan[i] != 0; }
+```
+
+See [Binary (de)serialization of structs with paddings or non-blittable members](#12-binary-deserialization-of-structs-with-paddings-or-non-blittable-members) for more information.
+
+## 22. Interop
+
+While most of the suggestions in this document apply to interop scenarios as well, it is recommended to follow the [Native interoperability best practices](https://learn.microsoft.com/dotnet/standard/native-interop/best-practices) guide. Additionally, consider using auto-generated interop wrappers like [CsWin32](https://github.com/microsoft/CsWin32) and [CsWinRT](https://github.com/microsoft/CsWinRT/). This minimizes the need for you to write manual interop code and reduces the risk of introducing memory safety issues.
+
+## 23. Thread safety
+
+See [Managed threading best practices](https://learn.microsoft.com/dotnet/standard/threading/managed-threading-best-practices) and [.NET Memory Model](https://github.com/dotnet/runtime/blob/main/docs/design/specs/Memory-model.md) for more details.
+
+## 24. Unsafe code around SIMD/Vectorization
+
+See [Vectorization guidelines](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/vectorization-guidelines.md) for more details.
+In the context of the unsafe code it's important to keep in mind:
+
+* SIMD operations have complex requirements to provide atomicity guarantees (sometimes, they don't provide them at all).
+* Most SIMD Load/Store APIs don't provide bounds checks.
+
+## 25. Fuzz testing
+
+Fuzz testing (or "fuzzing") is an automated software testing technique that involves providing invalid, unexpected, or random data as inputs to a computer program. It provides a way to detect memory safety issues in code that may have gaps in test coverage. Tools like [SharpFuzz](https://github.com/Metalnem/sharpfuzz) can be used to set up fuzz testing for .NET code. In the dotnet/runtime repository, the [fuzzlyn](https://github.com/jakobbotsch/Fuzzlyn) fuzzing tool reported more than 130 bugs against RyuJIT and other components.
```

Comment:
```suggestion
Fuzz testing (or "fuzzing") is an automated software testing technique that involves providing invalid, unexpected, or random data as inputs to a computer program. It provides a way to detect memory safety issues in code that may have gaps in test coverage. Tools like [SharpFuzz](https://github.com/Metalnem/sharpfuzz) can be used to set up fuzz testing for .NET code.
```
Mentioning Fuzlyn in customer facing docs is not helpful. We can mention success stories for sharpfuzz if there are any.